src/v/s3/client.h

```diff
@@ -67,10 +67,10 @@ ss::future<iobuf> drain_response_stream(http::client::response_stream_ref resp);
 struct configuration : net::base_transport::configuration {
     /// URI of the S3 access point
     access_point_uri uri;
-    /// AWS access key
-    public_key_str access_key;
-    /// AWS secret key
-    private_key_str secret_key;
+    /// AWS access key, optional if configuration uses temporary credentials
+    std::optional<public_key_str> access_key;
+    /// AWS secret key, optional if configuration uses temporary credentials
+    std::optional<private_key_str> secret_key;
```

Comment:
this config object is used as a temporary carrier of values from the application start to where the `cloud_storage_api` launches. 

these values are passed along to the cloud storage api (specifically `auth_refresh_bg_op::build_static_credentials` in `src/v/cloud_storage/remote.cc`) at start, and that object either:

1. does not use these values because credentials will be loaded from remote API calls OR
2. uses the values contained in optionals and passes them along to the credential appliers

the values in this config object are not used in the long term, these are made optional so that they can be absent when the application is configured to get the credentials from remote apis.